### ISSUE TITLE
rio-system po/socat pod keeps restarting after install

### ISSUE THREAD

**Issue by tfiduccia** [rancher/rio#315]
Version - v0.2.0-rc1

Steps:
1.  Install rio on mac using minikube
2.  kubectl get po -n rio-system

results:  socat pod keeps restarting
![image](https://user-images.githubusercontent.com/11514927/60136508-2e871900-9759-11e9-9223-b32cf3be2c45.png)

**Comment by milkowski** [comment]
It seems it cannot bind on start to requested address:
`2019/07/23 11:52:55 socat[1] E bind(5, {AF=2 127.0.0.1:5442}, 16): Address in use`

**Comment by betatim** [comment]
I have the same problem on a newly created minikube instance. Versions: OSX, minikube v1.0.0, k8s 1.14, rio-darwin-amd64 version v0.2.0 (cc7084b7).

Is there something else to try to help debug this?

**Comment by betatim** [comment]
One more bit of information: to install rio I executed: `rio install` which hangs/never completes and eventually displays:
```
...
\ Waiting for all the system components to be up. Not ready: [autoscaler build-controller kiali]a kiali registry]li registry]
Detecting that your service loadbalancer for service mesh gateway is still pending. Do you want to:
[1] Use HostPorts(Please make sure port 9080 and 9443 are open for your nodes)
[2] Wait for Service Load Balancer
```

**Comment by StrongMonkey** [comment]
I will take a look at this issue. @betatim For your case, I think your minikube version might be old that it doesn't support service load balancer(rio uses that expose istio-gateway to internal services). Using host-ports should work.

**Comment by betatim** [comment]
Thanks for the tip with the minikube version! That fixed it, now on minikube 1.2.0 and k8s 1.15 and it works as advertised!

The socat pod still struggles/is in CrashLoopBackOff.

**Comment by StrongMonkey** [comment]
CrashLoopBackOff is related to the socat container can't be launched with hostNetworking though, It is a different issue. But at this point it should not impact any existing functionality. @betatim Let me know if you are running into any issue though.

**Comment by cerebroso** [comment]
I have encountered this issue as well, when trying to install it in a kubernetes cluster (1.13.9-gke.3). The logs for the socat pods all say something like this:

`2019/09/11 20:49:31 socat[1] E bind(5, {AF=2 127.0.0.1:5442}, 16): Address in use`

**Comment by StrongMonkey** [comment]
This should be have been fixed in the latest v0.6 alpha release.

**Comment by rancher-max** [comment]
Appears to be working with minikube v1.5.2 and rio v0.6.0-alpha.2

After a few hours, there have been 0 restarts on socat.
